Gives Chief of Security hangar access & map edit.

zaredman opened the pull request:
What the title says. Gives ~~Security~~ COS hangar access. Also adds a button by the hangar so people inside who can't get out can get out, like what we have in Medical.

I know this was talked about previously on #21135 and numerous times both in-game and on Discord. That PR was back in April. Although I do understand the 'hangar creep' argument about this, I feel it's at a point where Sec needs it often enough.

I didn't set the doors to `access_solgov_crew`/`access_nanotrasen` because people like the Chef and Bartender, for example, still have no good reason to go into the hangar.

Green-00ff00 committed:
Gives Security access to the hangar.

Green-00ff00 committed:
Changelog.

CakeQ commented:
Yes please

chaoko99 commented:
Oh my god thank you.

Sbotkin commented:
Just give it access_solgov_crew then.

ghost commented:
>Just give it access_solgov_crew then.

>I didn't set the doors to access_solgov_crew/access_nanotrasen because people like the Chef and Bartender, for example, still have no good reason to go into the hangar.

Sbotkin commented:
Bartender doesn't have it. Chef won't go in hangar. This PR doesn't really makes sense because you give access to everyone basically.

chaoko99 commented:
but he isn't though

ghost commented:
>Bartender doesn't have it.

```
/datum/job/bartender
	department = "Service"
	department_flag = SRV
	supervisors = "the Executive Officer"
	ideal_character_age = 30
	selection_color = "#515151"
	outfit_type = /decl/hierarchy/outfit/job/torch/crew/service/bartender
	allowed_branches = list(/datum/mil_branch/civilian)
	allowed_ranks = list(/datum/mil_rank/civ/contractor)

	access = list(access_hydroponics, access_bar, access_solgov_crew, access_kitchen)
```

This is just for Security. If someone else wants to do one making it `access_solgov_crew` go on ahead.

AtomicWorm commented:
I don't like this. Why does security need hangar access? What business does security have in the hangar?

If security personnel need to get inside the hangar, then they can go ask somebody who actually has access to it because they actually have business being in there. There will usually be more than one such people around, such as a Deck Officer, Technician, miners.

chaoko99 commented:
"Oh no someone on the away site is killing people"

I will go ahead and concede that this is what the Aquila is for, though.

babydoll commented:
> As an aside, I don't have a horse in this race. I couldn't care if it goes through or not. Figured if people want it, here it is.

This PR needs a sponsor to argue for it, as the author doesn't care.

ghost commented:
>This PR needs a sponsor to argue for it, as the author doesn't care.

I mean, I can argue for it. I personally think it would be a very useful adjustment. I wouldn't be heartbroken if it didn't get in is what I meant by that. Today's example:

Lowpop. Sec ends up in the Hangar after an away mission. Happens to come back ~~during~~ just before a radiation storm. Gets completely rad'd out and almost dies from it all because no-one could open the door for him.

Yesterday? Had a not-an-actual-janitor hole himself up in the Charon and ended up breaking down a wall to try to get out. Security's response was delayed since no-one could get the door for them quickly.

Within recent history? Antag NT activating "Fort Science" in the Petrov, antag shoot-outs/other attempted murders in the hangar, explorers bringing back ships full of critters that want to kill everyone, etc. Traitor tot pilot/explorer/nt guys There's been a good number of times where Sec ends up in the hangar after someone gets gibbed by being in the LZ when they shouldn't have that Sec would end up investigating as a death, etc. I'm sure if you ask on Discord or in-game, people could rattle off more examples than I could as I don't sec all too often. But even with that, it seems like once a shift when I sec I need to get into the hangar for some reason.

My main gripe is that it's one of the main points of entry to the ship and the guys that are tasked with keeping the ship safe can't even get in there without hand-holding from someone else.

babydoll commented:
Isn't the door flanked by two windows? If it's really an emergency, break the window. It can be fixed. The window's a lot less valuable than crew. 

Security especially needs roadblocks in the way they move around the map to allow **other** people to interact and be interacted with before the guys with guns come in to (often) end it completely. Security deserves those opportunities too, but they already get the best chance out of most everyone.

This might let security do their job 'better' from a purely mechanical perspective, but I'm not sure if it's the best for everyone else.

ghost commented:
Alternative, and the basis for the fork name: Could also do one of those 'push buttons to exit from hangar to hall' down there like we have up in Medbay.

afterthought2 commented:
Just FYI: access_hangar gives access to certain deck management software functionality as well.

I'm opposed to giving sec hanger access on principle (yes, the hangar creep argument). Basically what sabiram said above.

chaoko99 commented:
"hangar creep", this is a pseudo-MUD, not an MMO. You aren't hurting anyone by allowing this.

I think the COS should have access and nobody else, if this helps any.

comma commented:
CoS having access is fine with me.
Add exit button if you are afraid of being stuck too sure.

ghost commented:
I’ll switcharoo it so it’s COS and button.

Green-00ff00 committed:
Adds exit button to hangar.

Removes hangar access from Master at Arms, Brig Officer, and Detective.

Changelog updates.

babydoll commented:
Needs to be squashed, please refer to [this guide](https://github.com/Baystation12/Baystation12/wiki/Squashing) or ask on IRC/discord if you need help.

Looks good, otherwise

ghost commented:
Think I did it right?
